packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ModelProvider.cs

```diff
@@ -356,5 +388,54 @@ private MethodBodyStatement GetPropertyInitializers(
 
             return rawDataField;
         }
+
+        private PropertyProvider? BuildAdditionalProperties()
+        {
+            var additionalProperties = _inputModel.AdditionalProperties;
+            if (additionalProperties is null)
+            {
+                return null;
+            }
+
+            var valueType = CodeModelPlugin.Instance.TypeFactory.CreateCSharpType(additionalProperties);
+            if (valueType is null)
+                throw new InvalidOperationException($"Failed to create CSharpType for additional properties of model {_inputModel.Name}");
+
+            var originalType = new CSharpType(typeof(IDictionary<,>), typeof(string), valueType);
+            var additionalPropsType = !_inputModel.Usage.HasFlag(InputModelTypeUsage.Input)
+                ? ReplaceUnverifiableType(originalType).OutputType
+                : ReplaceUnverifiableType(originalType);
+
+            return new PropertyProvider(
+                null,
+                MethodSignatureModifiers.Public,
+                additionalPropsType,
+                name: AdditionalPropertiesPropertyName,
+                new AutoPropertyBody(false),
+                enclosingType: this);
+        }
+
+        private CSharpType ReplaceUnverifiableType(CSharpType type)
```

Comment:
What does unverifiable mean in this context? Can we add some comments?

Reply:
Added some comments.